frontend/components/ClubEditPage/ClubEditCard.tsx

```diff
@@ -353,14 +378,88 @@ export default function ClubEditCard({
   const creationDefaults = {
     subtitle: 'Your Subtitle Here',
     email_public: true,
+    accepting_members: false,
   }
 
+  const editingFields = new Set()
+  fields.forEach(({ fields }) =>
+    fields.forEach(({ name }) => editingFields.add(name)),
+  )
+
   return (
-    <Form
-      fields={fields}
-      defaults={Object.keys(club).length ? club : creationDefaults}
+    <Formik
+      initialValues={
+        Object.keys(club).length
+          ? doFormikInitialValueFixes(
+              Object.entries(club).reduce((acc, [key, value]) => {
+                if (
+                  editingFields.has(key) ||
+                  (key.endsWith('_url') &&
+                    editingFields.has(key.substr(0, key.length - 4)))
+                ) {
+                  acc[key] = value
+                }
+                return acc
+              }, {}),
+            )
+          : creationDefaults
+      }
       onSubmit={submit}
-      enableSubmitWithoutEdit={true}
-    />
+      enableReinitialize
+    >
+      {({ dirty, isSubmitting }) => (
+        <FormFieldClassContext.Provider value="is-horizontal">
```

Comment:
Could this be factored out into a separate component? Could help with organization, not sure exactly what this block is without a name.